packages/block-editor/src/components/link-control/test/index.js

```diff
@@ -2056,6 +2056,26 @@ describe( 'Addition Settings UI', () => {
 			} )
 		);
 	} );
+
+	it( 'should show tooltip with full URL alongside filtered display', async () => {
+		const user = userEvent.setup();
+		const url = 'https://example.com';
+		render( <LinkControl value={ { url } } /> );
+
+		const link = screen.getByRole( 'link' );
+
+		expect( screen.getByRole( 'link' ) ).toHaveTextContent(
```

Comment:
We can totally go with the constant above instead of doing another query here. 

```suggestion
		expect( link ).toHaveTextContent(
```